src/coreclr/debug/ee/controller.cpp

```diff
@@ -79,27 +149,107 @@ SharedPatchBypassBuffer* DebuggerControllerPatch::GetOrCreateSharedPatchBypassBu
     }
     CONTRACTL_END;
 
-    if (m_pSharedPatchBypassBuffer == NULL)
+    _ASSERTE(m_pSharedPatchBypassBuffer == NULL);
+    if (m_pSharedPatchBypassBuffer != NULL)
```

Comment:
If we've already got the code for GetOrCreate() semantics you could remove the asserts, rename the method GetOrCreateSharedPatchBypassBuffer and delete the unnecessary GetSharedPatchBypassBuffer() function. We only needed separate Create and Get so that DebuggerPatchSkip knew when to init the memory. Now that the memory is inited here the constructor no longer needs to distinguish them.

Currently this code path isn't initializing InstructionAttribute*. You'd either need to add init code for it or in another comment I suggested having a SharedPatchBuffer::GetInstructionAttribute() method. Either way works.